AdvSearchPage.py

```diff
@@ -175,10 +179,15 @@ def entries(results, offset):
         searchterms = []
         for key in terms:
             if key in ['author', 'title', 'subject']:
-                for word in params[key].split():
-                    searchterms.append((key, word))
+                word_set = set()
+                for word in RE_SPLIT.split(params[key]):
+                    if len(word) > 1:
+                        word_set.add(word)
+                searchterms.extend([(key, word) for word in word_set][:MAX_WORDS])
```

Comment:
Ordering in sets is non-deterministic, so this isn't going to get the _first_ `MAX_WORDS`, it's just going to get _some_ `MAX_WORDS`. For real users this will result in non-deterministic search results if they put in more than 5 words which seems less than ideal.

```
>>> s=set()
>>> s.add(1)
>>> s.add(6)
>>> s.add(12)
>>> s
{1, 12, 6}
>>> b=[x for x in s]
>>> print(b)
[1, 12, 6]
```

Reply:
I noticed that. I think maybe ordering from longest word to shortest so that It doesn't start by searching "the" would be better.

Reply:
Ordering in dicts is determinstic, so you can use one to mimic an ordered set.

```
>>> d=dict()
>>> d["a"] = None
>>> d["b"] = None
>>> d["c"] = 3
>>> d["d"] = None
>>> d["c"] = None
>>> d
{'a': None, 'b': None, 'c': None, 'd': None}
>>> d.keys()
dict_keys(['a', 'b', 'c', 'd'])
>>> list(d.keys())
['a', 'b', 'c', 'd']
>>> for i in d.keys():
...     print(i)
...
a
b
c
d
```